docs/synchronization.md

```diff
@@ -0,0 +1,220 @@
+|--------|---------|
+| disableSyncData | (boolean) Disable storage of  synchronization data. Default: `false` |
+| findLegacyDocument | (function) Custom function which a konnector may use to look for a legacy document. Default is `null`. See below for more details.
+| getSyncData | (function) A mapping method returning additional synchronization data which have to be added to the document. |
+| idAttribute | (string&#124;Array&#124;function) How the identifier attribute is named in the entry. Default value: `id`. See below for additional customization |
+| konnector | *Mandatory* (string) The slug (or better, an uuid) of the current konnector.|
+| shouldSave | (function) A method taking current entry and matching document in database to evaluate if an entry has to be saved in database. See below default method and customization.|
+| shouldUpdate | (function) A method taking current entry and matching document in database to evaluate if a document must be updated with current entry. See below for default method and customization. |
+
+### findLegacyDocument
+If a document has not been saved yet with actual synchronization data, we will have no way to retrieve it, except by providing a `findLegacyDocument` function to `saveEntry` options. This signature of a `findLegacyDocument` function is:
+```js
+async function (entry, cozy) => { /* look for your document */ }
+```
+The `cozy` parameter is an instance of Cozy-Client.
+
+The idea is for example to retrive a document by its file path or any other accurate information.
+
+### getSyncData
+This method returns any addtional synchronization data the konnector should need. Default method should be something returning an empty object or `null`.
+```js
+(entry) => ({})
+```
+A konnector should provide its own method:
+```js
+{
+  getSyncData: (entry) => ({
+    filename: entry.name
+  })
+}
+```
+
+### idAttribute
+#### String
+Name of the id attribute in entry. Default is `id`. If we want to use another name, we may use:
+```js
+saveFiles(files, {idAttribute: 'uuid'})
+```
+#### Array
+This attribute may also be an array of string, defining each entry attribute which must be used for identifier.
+
+##### Example
+```js
+saveFiles(files, {idAttribute: ['name', 'author', 'date']})
+```
+
+In this case, all fields are used in synchronization data.
+
+
+#### function
+
+When an entry does not provide an unique identifer, it is possible to use a function as `idAttribute` to customize the way the unique identifer is generated.
+
+##### Example
+```js
+saveFiles(files, {idAttribute: (entry) => hash(entry)})
+// hash() is a custom method  hashing the current file.
+```
+
+### konnector
+
+The konnector attribute is used as a key reference to retrieve all document synchronized by this konnector.
+
+For now, the `konnector` option will be needed for every call. But it could be interesting to initialize some kind of runner with it, which may encapsulate all the `cozy-konnector-libs` functions.
```

Comment:
I think we can get the slug directly from the account (account_type) and feed saveFiles with it somehow.

Reply:
`account_type` is supposed to be removed, for now it is just maintained for legacy, as every information is now located in the trigger. As the sync data need the konnector slug, it feels natural to me to pass it as a parameter somewhere. What do you think ?